### ISSUE TITLE
Bug: Windows Version tries to upload to http://upload 

### ISSUE THREAD

**Issue by ILightThings** [patrickhener/goshs#23]
I think there may be an issues with translating between windows files system and linux file system. `/` and `\`.

When trying to upload, the post request get set to http://upload instead of the the userpath.

```
Host: upload
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:98.0) Gecko/20100101 Firefox/98.0
Accept: application/json
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Cache-Control: no-cache
X-Requested-With: XMLHttpRequest
Content-Type: multipart/form-data; boundary=---------------------------25573539524841597451864585240
Content-Length: 194636
Origin: http://192.168.1.159:8000
Connection: keep-alive
Referer: http://192.168.1.159:8000/
```

![firefox_EMZRbO1PB6](https://user-images.githubusercontent.com/27817827/157742039-323ceb92-b831-4e66-9474-b45bccd3bca3.png)

If you are busy, I can try to make a fix to it, but I would need to learn the code.

**Comment by patrickhener** [comment]
Oh hey. Yeah this could be. Never tried upload on Windows tbh. If you are able to fix go ahead and send me a pull request with a fix. Would be very welcome. Otherwise I might look at it if I have spare time.

Thanks for letting me know.

**Comment by patrickhener** [comment]
So changing path for filepath will only "repair" the display on the page though. Upload still breaks cause for some reason I am not yet aware of. For some reason on windows when upload the OPTIONS request in fact goes to "Host: upload" instead of "Host: <ip:port>"/ipload.

So I have to find out why.

**Comment by patrickhener** [comment]
```
    <!-- Dropzone related config -->
    <script>
        Dropzone.autoDiscover = false;
    </script>
    {{ if (eq .Directory.RelPath "/") }}
        <script>
            let url = "/upload"
        </script>
    {{ else }}
        <script>
            let url = "{{.Directory.RelPath}}/upload"
        </script>
    {{ end }}

    <script>
        let myDropzone = new Dropzone("div#mydropzone", {
            autoProcessQueue: false,
            paramName: "files",
            method: "post",
            url: url,
            uploadMultiple: true,
            parallelUploads: 100,
            maxFiles: 100,
            maxFilesize: 10240,

            init: function () {
                document.querySelector("button[type=submit]#submit-dropzone").addEventListener("click", function(e) {
                    e.preventDefault();
                    e.stopPropagation();
                    myDropzone.processQueue();
                })

            }
        });

        myDropzone.on("successmultiple", function () {
            location.reload();
        });
    </script>
```
This is from `index.html` in static template folder. So in fact in linux it seams like you are in the root of where goshs is running when posting to `/upload` it will take this as path and add the ip as host.

This behaves different in windows it seams. It will take /upload as host instead of path there.

Working scenario would be to upload a file in a subdirectory while using windows. This is just a workaround though.

**Comment by patrickhener** [comment]
Fixed with commit 23631a8
